server/sql.go

```diff
@@ -208,6 +208,25 @@ func (s *Server) registerSQLRoutes(g *echo.Group) {
 					}
 				}
 			}
+
+			// Check database access rights.
+			principalID := c.Get(getPrincipalIDContextKey()).(int)
```

Comment:
We can use ProjectPermission for this.

Reply:
I think we cannot use `ProjectPermission` here. Database access control relays on the `policy` module.